test/integration/query-interface/changeColumn.test.js

```diff
@@ -284,6 +284,62 @@ describe(Support.getTestDialectTeaser('QueryInterface'), () => {
         expect(table.name.unique).to.equal(true, '(2) name column should remain unique');
       });
 
+      it('should not add single column unique constraints from multi-column constraint when adding or modifying columns', async function() {
```

Comment:
Added a new test to ensure multi-column indexes to not turn into single column indexes.